File: 189_done.cpp
/*correct and easy approach to rotate a array list etc*/
#include<bits/stdc++.h>
#include<iostream>

using namespace std;

int rotate(vector<int>& nums,int k){
  int n=nums.size();
  k = k%n;

  //reversing the whole array 
  reverse(nums.begin(),nums.end());
  // reversing the array from begin upto k
  reverse(nums.begin(),nums.begin()+k);
  // reversing the array from k to end
  reverse(nums.begin()+k,nums.end());

}

int main()
{
  vector<int> nums({1,2,3,4,5});
  int k=1;
  rotate(nums,k);
  return 0;
}

File: 189done.py
# trying to rotate a list
def rotate(lst,k):
    n = len(lst)
    k=k%n

    # reversing the whole lst
    lst[::-1]
    # reversing the element only up to k 
    lst[:k:-1]
    # reversing the element from k to end
    lst[k::-1]
    return lst

k=1
lst=[1,2,3,4,5]
rotate(lst,k)

